### PR TITLE
Implemented aws-sdk-swift package integration and removed amplify

### PR DESCRIPTION
**Pull request by zeeshanmakeen**
**[Important Merge Auth SDK [PR](https://github.com/aws-geospatial/amazon-location-mobile-auth-sdk-ios/pull/7) first]**
- Implemented aws-sdk-swift package integration
- Removed all amplify integration
- Some refactoring and optimization
- Some bug fixes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,6 @@
 .DS_Store
+*.xcworkspacedata
+Tests/AWSTrackingSDKiOSTests/TestConfig.plist
+.swiftpm/xcode/package.xcworkspace/xcuserdata/zeeshansheikh.xcuserdatad/UserInterfaceState.xcuserstate
```
Comment (ejboucher) [inline .gitignore:4]: Let's avoid merging any changes that are specific to your own username.

File: Package.resolved [hunk 2]
```diff
@@ -3,19 +3,28 @@
     {
       "identity" : "amazon-location-mobile-auth-sdk-ios",
       "kind" : "remoteSourceControl",
-      "location" : "https://github.com/aws-geospatial/amazon-location-mobile-auth-sdk-ios.git",
+      "location" : "https://github.com/makeen-project/amazon-location-mobile-auth-sdk-ios.git",
```
Comment (ejboucher) [inline Package.resolved:6]: We should not change this - we need to target the official repository.

**Review by ejboucher — CHANGES_REQUESTED** [review]
We need to make a few changes before this can be merged - please see comments.

File: Package.resolved (hunk 2 above)
Comment (olegfilimonov) [reply Package.resolved]: Updated

File: .gitignore (hunk 1 above)
Comment (olegfilimonov) [reply .gitignore]: Gitignore updated

**Review by jonayuan — APPROVED** [review]

**Pull request merged**